cyclomatic_complexity/api_generator.py

```diff
@@ -51,11 +63,18 @@ def make_script_view(script, desired_file_ext=""):
 
     body = (
         *api_fs_exec_utils.generate_exec_header(), r"",
+        *api_fs_bash_utils.generate_extract_attr_value_from_string()[1], r"",
+        *api_fs_bash_utils.add_suffix_if_exist()[1], r"",
+        *api_fs_bash_utils.wait_until_pipe_exist()[1], r"",
         *api_fs_exec_utils.generate_get_result_type(file_extension), r"",
         *api_fs_exec_utils.generate_api_node_env_init(), r"",
+        api_fs_bash_utils.generate_extract_attr_value_from_string()[0] + " \"SESSION_ID\" ${2} \"\" '=' SESSION_ID_VALUE", r"",
+        api_fs_bash_utils.add_suffix_if_exist()[0] + " \"${SESSION_ID_VALUE}\" \"result.xml\" MULTISESSION_PIPE_OUT", r"",
         *api_fs_exec_utils.generate_read_api_fs_args(), r"",
         r'echo "${IN_ARGS[@]}" > ${SHARED_API_DIR}/${MAIN_SERVICE_NAME}/cc/statistic/GET/exec',
-        r"cat ${SHARED_API_DIR}/${MAIN_SERVICE_NAME}/cc/statistic/GET/result.xml | ${WORK_DIR}/pmccabe_visualizer/collapse.py ${brr[@]}",
+        api_fs_bash_utils.wait_until_pipe_exist()[0] + " ${SHARED_API_DIR}/${MAIN_SERVICE_NAME}/cc/statistic/GET/${MULTISESSION_PIPE_OUT}",
+        r"cat ${SHARED_API_DIR}/${MAIN_SERVICE_NAME}/cc/statistic/GET/${MULTISESSION_PIPE_OUT} | ${WORK_DIR}/pmccabe_visualizer/collapse.py ${brr[@]}",
```

Comment:
not that easy: the last argument `$brr{@}` must be declared in a context of a generated function, which requiry to pass exactly the same argument and declare it by exactly the same name in that function. Having different ${brr[@]} for different functions didn't adds up the idea of universal function, because argument name must be unique per function: somehere it is `brr`, somewhere is it quite different